actors/test/actor-tests/src/test/java/cloud/orbit/actors/test/DeactivationTest.java

```diff
@@ -63,6 +63,7 @@ public void simpleCleanupTest() throws ExecutionException, InterruptedException
         final UUID id = actor1.getUniqueActivationId().join();
         clock.incrementTime(20, TimeUnit.MINUTES);
         stage.cleanup().join();
+        waitForDeactivations();
```

Comment:
This change was a conscious decision to simplify the flow. The only place the behaviour matters is tests because the simulate time and expect the deactivation to have happened upon cleanup completing.